dotnet/src/SemanticKernel.Core/Planning/Plan.cs

```diff
@@ -248,6 +254,11 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
             // Execute the step
             var result = await context.Runner.RunAsync(step, functionVariables, cancellationToken).ConfigureAwait(false);
 
+            if (result is PlanResult planResult)
+            {
+                this.StepResults.Add(planResult);
+            }
```

Comment:
What is the case where this is not a PlanResult and the result can be ignored?

Reply:
This is the case when method `InvokeNextStepAsync` is invoked directly or through `kernel.StepAsync` extension method. It's the ability to control plan execution manually. In this case, `context.Runner` may run a function, which will return not `PlanResult`, but `FunctionResult`, which means that this Plan step doesn't have any child steps. In this case we should not add it to `StepResults` property and this collection will be empty.